docs/sdk/client-side-sdks/nextjs/nextjs-gettingstarted.md

```diff
@@ -0,0 +1,268 @@
+[![GitHub](https://img.shields.io/github/stars/devcyclehq/js-sdks.svg?style=social&label=Star&maxAge=2592000)](https://github.com/devcyclehq/js-sdks)
+
+
+## Usage (App Router)
+### Create the DevCycle context and export it
+To use DevCycle on the server, you must create a context that can be shared across your server components. The context
+will hold the user data and configuration for the current request, and ensures subsequent calls to retrieve variables
+are scoped to that data.
+
+In a shared file somewhere (for example, `app/devcycle.ts`):
+```typescript
+import { setupDevCycle } from '@devcycle/nextjs-sdk/server'
+
+const getUserIdentity = async () => {
+    // pseudocode function representing some call you might make to 
+    // your code to determine the current user
+    // You can use Next APIs such as `headers()` and `cookies()` here
+    const myUser = await determineUserIdentity(cookies())
+    return {
+        user_id: myUser.id
+    }
+}
+
+const { getVariableValue, getClientContext } = setupDevCycle(
+    // SDK Key. This will be public and sent to the client, so you MUST use the client SDK key.
+    process.env.NEXT_PUBLIC_DEVCYCLE_CLIENT_SDK_KEY ?? '',
+    // pass your method for getting the user identity
+    getUserIdentity,
+    // pass any options you want to use for the DevCycle SDK
+    {
+        enableStreaming: false,
+    },
+)
+
+export { getVariableValue, getClientContext }
+```
+Provide the context function to the DevCycleClientsideProvider as high as possible in your component tree.
+
+```typescript jsx
+import { DevCycleClientsideProvider } from '@devcycle/nextjs-sdk'
+// import the getClientContext method from your shared DevCycle file
+import { getClientContext } from './devcycle'
+
+export default async function RootLayout({
+    children,
+}: {
+    children: React.ReactNode
+}) {
+    return (
+        <html lang="en">
+            <body>
+                <DevCycleClientsideProvider
+                    context={await getClientContext()}
+                >
+                    {children}
+                </DevCycleClientsideProvider>
+            </body>
+        </html>
+    )
+}
+```
+Note: You _must_ use the client SDK key of your project, not the server SDK key. The key is used across the server and
+the client and will be sent to the clientside to bootstrap the client SDK.
+
+The setupDevCycle method will:
+- provide a getVariableValue method that encapsulates your configured SDK key, user getter and options
+- fetch your project's configuration from DevCycle when needed
+- return a context to be passed to the client component provider that provides a clientside DevCycle SDK, and bootstraps
+  it with the server's user and DevCycle configuration data.
+
+It will also await the retrieval of the DevCycle configuration, thus blocking further rendering until the flag states
+have been retrieved and rendering can take place with the correct values.
+
+### Get a variable value (server component)
```

Comment:
Should these sections be moved to a `Usage` page like React is split up? Also did we add the `Getting All Features / Variables` methods or any other methods not captured here?

Reply:
I dunno about splitting it up, it feels a bit disjointed to not be able to immediately see how to use it